src/GradeBook.tsx

```diff
@@ -59,12 +60,25 @@ function GradeBook() {
 			.catch(() => {
 				setSignedIn(false);
 			});
-	});
+	}, []);
+
+	useEffect(() => {
+		if (signedIn || !signedIn) {
+			getCurrentSemester()
+				.then((semester) => {
+					setSemester(semester);
+				})
+				.catch(() => {
+					console.error('Semester Not Found');
+				});
+		}
+	}),
+		[signedIn];
 
 	return (
 		<GradeBookContainer>
 			<Header text="Passr" />
-			<SideBar currentSemester={Fall2020} />
+			{semester && <SideBar currentSemester={semester} />}
```

Comment:
We should modify the sidebar to handle the case where it doesn't have a semester.